server/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/PlainHighlighter.java

```diff
@@ -157,10 +170,15 @@ public HighlightField highlight(FieldHighlightContext fieldContext) throws IOExc
             }
         }
 
-        // fragments are ordered by score by default since we add them in best
-        if (field.fieldOptions().scoreOrdered() == false) {
-            CollectionUtil.introSort(fragsList, (o1, o2) -> o1.getFragNum() - o2.getFragNum());
```

Comment:
Also just curious about the switch from Lucene CollectionUtil to plain list sort here, any pros / cons about the former?

Reply:
Just that it means we get to use built-in collections; it's possible that lucene's introSort is slightly more efficient, but we're talking tiny lists and small sets of results so I thought clarity was more important.